src/semanticlayertools/linkage/cocitation.py

```diff
@@ -0,0 +1,171 @@
+            chunk_size = int(data.shape[0] / self.numberProc)
+            if chunk_size == 0:  # Deal with small data samples.
+                chunk_size = 1
+            chunks = np.array_split(data, chunk_size)
+            pool = multiprocessing.Pool(processes=self.numberProc)
+            cocitations = pool.map(self.getCombinations, chunks)
+            cocitCounts = Counter([x for y in cocitations for x in y])
+            sortCoCitCounts = [
+                (x[0][0], x[0][1], x[1]) for x in cocitCounts.most_common()
+            ]
+            tempG = ig.Graph.TupleList(sortCoCitCounts, weights=True, vertex_name_attr='id')
+            components = tempG.components()
+            sortedComponents = sorted(
+                [(x, len(x), len(x) * 100 / len(tempG.vs)) for x in components], key=lambda x: x[1], reverse=True
+            )
+            with open(os.path.join(self.outpath, infilename + '_graphMetadata.txt'), 'w') as outfile:
+                outfile.write(f'Graph derived from {filepath}\nSummary:\n')
+                outfile.write(tempG.summary() + '\n\nComponents (ordered by size):\n\n')
+                for idx, elem in enumerate(sortedComponents):
+                    gcompTemp = tempG.vs.select(elem[0]).subgraph()
+                    outfile.write(
+                        f"{idx}:\n\t{elem[1]} nodes ({elem[2]:.3f}% of full graph)\n\t{len(gcompTemp.es)} edges ({len(gcompTemp.es)*100/len(tempG.es):.3f}% of full graph)\n\n"
+                    )
+                    if idx == 0:
+                        gcouttuple = (
+                            elem[1],
+                            elem[2],
+                            len(gcompTemp.es),
+                            len(gcompTemp.es) * 100 / len(tempG.es)
+                        )
+            giantComponent = sortedComponents[0]
+            giantComponentGraph = tempG.vs.select(giantComponent[0]).subgraph()
+            giantComponentGraph.write_pajek(
+                os.path.join(self.outpath, infilename + '_GC.net')
+            )
+            with open(os.path.join(self.outpath, infilename + '.ncol'), 'w') as outfile:
+                for edge in sortCoCitCounts:
+                    outfile.write(f"{edge[0]} {edge[1]} {edge[2]}\n")
+        except Exception:
+            raise
+        if self.debug == "l2":
+            print(f'\tDone in {time.time() - starttime} seconds.')
+        return gcouttuple
+
+    def processFolder(self):
+        """Calculate cocitation for all files in folder."""
+        starttime = time.time()
+        with open(
+            os.path.join(
+                self.outpath, 'Giant_Component_properties.csv'
+            ), 'w'
+        ) as gcmetafile:
+            gcmetafile.write('year,nodes,nodespercent,edges,edgepercent\n')
+            for file in tqdm(os.listdir(self.inpath), leave=False):
+                try:
+                    year = re.findall(r'\d{4}', file)[0]
+                except Exception:
+                    raise
```

Comment:
I guess the code has a bunch of `except Exception: raise` because you used the except clauses for placing breakpoints.

Anyway, `raise ValueError(f"Can't find year in file {file}")` can really help users out when this exception is raised.